python/BioSimSpace/Sandpit/Exscientia/Parameters/_parameters.py

```diff
@@ -1111,3 +752,195 @@ def _validate(molecule=None, tolerance=None, max_distance=None,
     if property_map is not None:
         if not isinstance(property_map, dict):
             raise TypeError("'property_map' must be of type 'dict'")
+
+# Wrapper function to dynamically generate functions with a given name.
+# Here "name" refers to the name of a supported AMBER protein force field.
+import sys as _sys
+_namespace = _sys.modules[__name__]
+def _make_amber_protein_function(name):
```

Comment:
Can't we just use `functools.partial` for these decorators?

Reply:
Good point. I just reproduced the OpenFF approach, but obviously partials works in both caes :man_facepalming:

Reply:
Actually, I did it this way so that the docstrings were exposed, but I think I can use `functools.wraps` for that. Will see if I can get it to work.

Reply:
Hmm, I can't get things to work nicely both in the console and using Sphinx. I'll try a bit more, but might just keep my original approach as it avoids a whole bunch of dosctring hacking.

Reply:
Okay, I'm going to keep things as is with some minor modifications. My approach gives you the correct function signature and docstring for the console and Sphinx. Using `parial` and `update_wrapper` doesn't do this, i.e. the pre-filled parameter will still appear in the docs, which is confusing to the user.